src/tool/subcommands/api_cmd/generate_test_snapshot.rs

```diff
@@ -0,0 +1,265 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::*;
+use crate::{
+    blocks::{CachingBlockHeader, TipsetKey},
+    chain::ChainStore,
+    chain_sync::{network_context::SyncNetworkContext, SyncConfig, SyncStage},
+    daemon::db_util::load_all_forest_cars,
+    db::{
+        db_engine::open_db, parity_db::ParityDb, EthMappingsStore, MemoryDB, SettingsStore,
+        SettingsStoreExt, CAR_DB_DIR_NAME,
+    },
+    genesis::{get_network_name_from_genesis, read_genesis_header},
+    libp2p::{NetworkMessage, PeerManager},
+    libp2p_bitswap::{BitswapStoreRead, BitswapStoreReadWrite, Block64},
+    message_pool::{MessagePool, MpoolRpcProvider},
+    networks::ChainConfig,
+    shim::address::CurrentNetwork,
+    state_manager::StateManager,
+    KeyStore, KeyStoreConfig,
+};
+use fvm_shared4::address::Network;
+use openrpc_types::ParamStructure;
+use parking_lot::RwLock;
+use rpc::{eth::filter::EthEventHandler, RPCState, RpcMethod as _};
+use tokio::{sync::mpsc, task::JoinSet};
+
+pub async fn run_test_with_dump(
+    test_dump: &TestDump,
+    db: Arc<ReadOpsTrackingStore<ManyCar<ParityDb>>>,
+    chain: &NetworkChain,
+) -> anyhow::Result<()> {
+    if chain.is_testnet() {
+        CurrentNetwork::set_global(Network::Testnet);
+    }
+    let mut run = false;
+    let chain_config = Arc::new(ChainConfig::calibnet());
```

Comment:
The chain config should depend on the `chain` parameter, no?

Reply:
Good catch, fixed!